pkg/sqlcache/db/client.go

```diff
@@ -353,27 +358,43 @@ func closeRowsOnError(rows Rows, err error) error {
 
 // NewConnection checks for currently existing connection, closes one if it exists, removes any relevant db files, and opens a new connection which subsequently
 // creates new files.
-func (c *client) NewConnection() error {
+func (c *client) NewConnection(useTempDir bool) (string, error) {
 	c.connLock.Lock()
 	defer c.connLock.Unlock()
 	if c.conn != nil {
 		err := c.conn.Close()
 		if err != nil {
-			return err
+			return "", err
 		}
 	}
-	err := os.RemoveAll(InformerObjectCacheDBPath)
-	if err != nil {
-		return err
+	if !useTempDir {
+		err := os.RemoveAll(InformerObjectCacheDBPath)
+		if err != nil {
+			return "", err
+		}
 	}
 
 	// Set the permissions in advance, because we can't control them if
 	// the file is created by a sql.Open call instead.
-	if err := touchFile(InformerObjectCacheDBPath, informerObjectCachePerms); err != nil {
-		return nil
+	var dbPath string
+	if useTempDir {
+		dir := os.TempDir()
+		f, err := os.CreateTemp(dir, InformerObjectCacheDBPathRoot)
+		if err != nil {
+			return "", err
+		}
+		path := f.Name()
+		dbPath = path + ".db"
+		f.Close()
+		os.Remove(path)
```

Comment:
Isn't this deleting a file that does not necessarily exist? Looks like we are deleting a file at `<dir>/<InformerObjectCacheDBPathRoot> + .db` but we created a file at `<dir>/<InformerObjectCacheDBPathRoot>`?